packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -875,6 +1062,9 @@ function rerenderState<S>(
 }
 
 function pushEffect(tag, create, destroy, deps) {
+  if (enableSpeculativeWork && tag & HookHasEffect) {
+    markWorkInProgressPushedEffect();
```

Comment:
this is used by a certain kind of bailout where even if we called the function component we can still bailout if no observable effects were pushed